test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/TagHelperUsageDescriptorFactoryTest.cs

```diff
@@ -0,0 +1,385 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#if !DNXCORE50
+using System;
+using System.IO;
+using System.Reflection;
+using Microsoft.AspNet.Razor.TagHelpers;
+using Microsoft.AspNet.Testing;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNet.Razor.Runtime.TagHelpers
+{
+    public class TagHelperUsageDescriptorFactoryTest
+    {
+        private const string TestTypeSummarySuffix = "The summary for <see cref=\"T:Microsoft.AspNet.Razor.Runtime." +
+            "TagHelpers.DocumentedTagHelper\" />.";
+        private const string TestTypeRemarksSuffix = "Inherits from <see cref=\"T:Microsoft.AspNet.Razor.Runtime." +
+            "TagHelpers.TagHelper\" />.";
+        private const string TestSummaryPropertySuffix = "This <see cref=\"P:Microsoft.AspNet.Razor.Runtime." +
+            "TagHelpers.DocumentedTagHelper.SummaryProperty\" /> is of type <see cref=\"T:System.String\" />.";
+        private const string TestRemarksPropertySuffix = "The <see cref=\"P:Microsoft.AspNet.Razor.Runtime." +
+            "TagHelpers.DocumentedTagHelper.SummaryProperty\" /> may be <c>null</c>.";
+        private const string TestRemarksAndSummaryPropertySummarySuffix = "This is a complex <see cref=\"T:System." +
+            "Collections.Generic.IDictionary`2\" />.";
+        private const string TestRemarksAndSummaryPropertyRemarksSuffix = "<see cref=\"P:Microsoft.AspNet.Razor." +
+            "Runtime.TagHelpers.DocumentedTagHelper.SummaryProperty\" /><see cref=\"P:Microsoft.AspNet.Razor.Runtime" +
+            ".TagHelpers.DocumentedTagHelper.RemarksProperty\" />";
```

Comment:
"Test" is noise, "Suffix" is misleading, and word order is confusing. So `TypeSummary`, `TypeRemarks`, `PropertySummary` (`PropertyWithSummary_Summary` seems a tad redundant), `PropertyRemarks`, `PropertyWithSummaryAndRemarks_Summary`, `PropertyWithSummaryAndRemarks_Remarks`